src/Controller/ActivityPub/Magazine/MagazineModeratorsController.php

```diff
@@ -4,16 +4,60 @@
 
 namespace App\Controller\ActivityPub\Magazine;
 
+use App\Entity\Contracts\ActivityPubActivityInterface;
+use App\Entity\Magazine;
+use App\Entity\Moderator;
+use App\Repository\MagazineRepository;
+use App\Service\ActivityPubManager;
+use JetBrains\PhpStorm\ArrayShape;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Uid\Uuid;
 
 class MagazineModeratorsController
 {
-    public function __construct()
+    public function __construct(
+        private readonly ActivityPubManager $manager,
+        private readonly MagazineRepository $magazineRepository,
+        private readonly UrlGeneratorInterface $urlGenerator
+    ) {
+    }
+
+    public function __invoke(Magazine $magazine, Request $request): JsonResponse
     {
+        $data = $this->getCollectionItems($magazine);
+        $response = new JsonResponse($data);
+        $response->headers->set('Content-Type', 'application/activity+json');
+
+        return $response;
     }
 
-    public function __invoke(): JsonResponse
+    #[ArrayShape([
+        '@context' => 'string',
+        'type' => 'string',
+        'id' => 'string',
+        'totalItems' => 'int',
+        'orderedItems' => 'array',
+    ])]
+    private function getCollectionItems(Magazine $magazine): array
     {
-        return new JsonResponse();
+        $moderators = $this->magazineRepository->findModerators($magazine, perPage: $magazine->moderators->count());
+
+        $items = [];
+        foreach ($moderators->getCurrentPageResults() as /* @var Moderator $mod */ $mod) {
+            $actor = $mod->user;
+            $items[] = $this->manager->getActorProfileId($actor);
+        }
+
+        $id = Uuid::v4()->toRfc4122();
+
+        return [
+            '@context' => ActivityPubActivityInterface::CONTEXT_URL,
+            'type' => 'OrderedCollection',
+            'id' => $this->urlGenerator->generate('ap_object', ['id' => $id], UrlGeneratorInterface::ABSOLUTE_URL),
```

Comment:
I notice this evaluates a little differently for us than lemmy:

```
"id": "http://127.0.0.1/f/object/a6cb5314-87e5-46d2-9555-513c08bfc15f",
```

vs

```
"id": "https://lemmy.world/c/memes/moderators",
```

Reply:
you're right. I switched it to a generated id so it wouldn't be cached, but now that I clear the cache properly I should be able to change it back